app/services/duplicate_canonical_profile_resolver.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Class that deduplicates canonical profiles with identical benchmark/ref IDs
+class DuplicateCanonicalProfileResolver
+  class << self
+    def run!
+      duplicates.each do |duplicate|
```

Comment:
Pretty much always use `find_each`. It batches by 1000 by default so we don't run out of memory building and storing 1000s of profile objects in memory.

```suggestion
      duplicates.find_each do |duplicate|
```